modules/common/secret/secret.go

```diff
@@ -385,3 +385,42 @@ func GetDataFromSecret(
 
 	return data, ctrl.Result{}, nil
 }
+
+// VerifySecret - verifies if the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func VerifySecret(
+	ctx context.Context,
+	secretName types.NamespacedName,
+	expectedFields []string,
+	reader client.Reader,
+	requeueTimeout time.Duration,
+) (string, ctrl.Result, error) {
+	secret := &corev1.Secret{}
+	err := reader.Get(ctx, secretName, secret)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return "",
+				ctrl.Result{RequeueAfter: requeueTimeout},
+				fmt.Errorf("OpenStack secret %s not found", secretName)
+		}
+		return "", ctrl.Result{}, err
```

Comment:
while it might be not that important in this small func, I think we should wrap the errors we return to be easier to identify when debugging: `fmt.Errorf("get secret %s failed: %w", secretName, err)`. using `%w` makes it available to errors.Is and errors.As

Reply:
+1 lets wrap the errors